File: emotif/emotif/backend/Gemini.py
import google.generativeai as genai

genai.configure(api_key="AIzaSyBeKx1KioTFlKyEWeR2pFTZSCmtE5SuJpk")
model = genai.GenerativeModel("gemini-1.5-flash")
emotion = '{[null,{"date": "20-01-2025","emotion": "joy","level": 5,"time": "10:47"},{"date": "20-01-2025","emotion": "anxiety","level": 4,"time": "12:55"}]}'
events ='{[null,{"date": "20-01-2025","endtime": "19:00","name": "WiTS meeting","starttime": "18:00"},{"date": "20-01-2025","endtime": "12:25","name": "CSCI4147 Class","starttime": "11:35"},{"date": "20-01-2025","endtime": "15:55","name": "CSCI 4176","starttime": "14:35"},{"date": "20-01-2025","endtime": "10:30","name": "Gym","starttime": "9:30"}]}'
response = model.generate_content("Given the calendar events of a user:"+events+" and their logged emotions for a day:"+emotion+", could you create suggestions for activities that the user could do in their free time that would make the user feel better?Please provide your answer in the following format:{\"date\": \"\",\"endtime\": \"\",\"name\": \"\",\"starttime\": \"\"}")
print(response.text)

File: emotif/emotif/backend/Gemini.js
const { GoogleGenerativeAI } = require("@google/generative-ai");

async function askGemini(events, emotions) {
    try {
        const genAI = new GoogleGenerativeAI("AIzaSyBeKx1KioTFlKyEWeR2pFTZSCmtE5SuJpk");
        const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });

        const prompt = "Given the calendar events of a user:" +JSON.stringify(events)+" and their logged emotions for a day:" +JSON.stringify(emotions)+", could you create suggestions for activities that the user could do in their free time that would make the user feel better? Specifically say what activity and the reason for suggesting that activity. Please provide your answer in the following format:{\"date\": \"\",\"endtime\": \"\",\"name\": \"AI suggested activity: \",\"starttime\": \"\"}. Here is an example output:[{\"date\": \"20-01-2025\", \"endtime\": \"14:35\", \"name\": \"AI suggested activity: Go for a walk, it will make you feel less anxious\", \"starttime\": \"12:55\"}].Please do NOT include '```json' at start and '```' at the end of the output.";

        const result = await model.generateContent([prompt]);
       return result.response.text();
    } catch (error) {
        return 'Error generating content:'+ error;
    }
}
